lib/core/posts/details/src/utils/tall_media_classifier.dart

```diff
@@ -0,0 +1,129 @@
+// Flutter imports:
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+// Project imports:
+import '../../../../settings/settings.dart';
+
+class TallMediaDisposition {
+  const TallMediaDisposition._({
+    required this.isTall,
+    required this.shouldFitToWidth,
+    required this.aspectRatio,
+    required this.viewportRatio,
+    required this.pixelCount,
+    required this.scrollExtent,
+  });
+
+  const TallMediaDisposition.standard({
+    required double aspectRatio,
+    required double viewportRatio,
+    required double pixelCount,
+  }) : this._(
+         isTall: false,
+         shouldFitToWidth: false,
+         aspectRatio: aspectRatio,
+         viewportRatio: viewportRatio,
+         pixelCount: pixelCount,
+         scrollExtent: 0,
+       );
+
+  factory TallMediaDisposition.tall({
+    required double aspectRatio,
+    required double viewportRatio,
+    required double pixelCount,
+    required double scrollExtent,
+  }) => TallMediaDisposition._(
+    isTall: true,
+    shouldFitToWidth: true,
+    aspectRatio: aspectRatio,
+    viewportRatio: viewportRatio,
+    pixelCount: pixelCount,
+    scrollExtent: scrollExtent,
+  );
+
+  final bool isTall;
+  final bool shouldFitToWidth;
+  final double aspectRatio;
+  final double viewportRatio;
+  final double pixelCount;
+  final double scrollExtent;
+
+  bool get hasScrollableExtent => scrollExtent > 0;
+}
+
+TallMediaDisposition classifyTallMedia({
```

Comment:
- this should be a factory method in TallMediaDisposition, rename it to fromViewport. Move all calculation to this method
- settings.enabled check, it should be handle from the UI